packages/cli/src/prompts/choices.ts

```diff
@@ -0,0 +1,61 @@
+import { Separator } from 'inquirer';
+
+import difference from 'lodash.difference';
+import flatten from 'lodash.flatten';
+import ProjectFile from '../models/files/ProjectFile';
+import ContractManager from '../models/local/ContractManager';
+import Dependency from '../models/dependency/Dependency';
+
+export type Choice = string | { name: string; value: string } | { type: 'separator' };
+
+export type ContractsSource = 'built' | 'notAdded' | 'added' | 'all';
+
+// Generate a list of contracts names
+export function contracts(source: ContractsSource = 'built'): Choice[] {
+  const localProjectFile = new ProjectFile();
+  const contractManager = new ContractManager(localProjectFile);
+
+  const contractsFromBuild: Choice[] = contractManager.getContractNames();
+  const contractsFromLocal = Object.keys(localProjectFile.contracts)
+    .map(alias => ({ name: localProjectFile.contracts[alias], alias }))
+    .map(({ name: contractName, alias }) => {
+      const label = contractName === alias ? alias : `${alias}[${contractName}]`;
+      return { name: label, value: alias };
+    });
+
+  // get contracts from `build/contracts`
+  if (source === 'built') {
+    return contractsFromBuild;
+  }
+
+  // get contracts from project.json file
+  if (source === 'notAdded') {
+    return difference(
+      contractsFromBuild,
+      contractsFromLocal.map(({ value }) => value),
+    );
+  }
+
+  if (source === 'added') {
+    return contractsFromLocal;
+  }
+
+  // generate a list of built contracts and package contracts
+  if (source === 'all') {
```

Comment:
I totally avoid them because I never remember which languages have fallthrough semantics, and the possibility of forgetting to put in `break` makes me panic. We may be able to add a linter rule to force always using `break`...

In this case though I moved most of this code from `prompts.contractsList` and it was using `if` statements.

Reply:
Thanks for clarifications. Just want to mention that [airbnb eslint config](https://eslint.org/docs/rules/no-fallthrough)  doesn't allow [fallthrough](https://eslint.org/docs/rules/no-fallthrough).